cpp/src/arrow/util/tracing_internal.h

```diff
@@ -146,6 +210,22 @@ opentelemetry::trace::StartSpanOptions SpanOptionsWithParent(
         return st;                                                                \
       })
 
+#define GET_CURRENT_SPAN(lhs) \
+  lhs = ::arrow::internal::tracing::GetTracer()->GetCurrentSpan()
+
+#define SET_SPAN_SCOPE(lhs, span) \
+  lhs = ::arrow::internal::tracing::GetTracer()->WithActiveSpan(span)
+
+#define TIE_SPAN_TO_GENERATOR(generator) \
+  generator = ::arrow::internal::tracing::TieSpanToAsyncGenerator(std::move(generator))
+
+#define PROPAGATE_SPAN_TO_GENERATOR(generator)                                \
+  generator = ::arrow::internal::tracing::PropagateSpanThroughAsyncGenerator( \
+      std::move(generator))
+
+#define WRAP_ASYNC_GENERATOR(generator, name) \
+  generator = ::arrow::internal::tracing::WrapAsyncGenerator(std::move(generator), name)
+
```

Comment:
What do we gain making these macros instead of static methods?  If the namespaces are troublesome then lets just put the methods in the `arrow` namespace.  That seems a better compromise than introducing a macro.

```
namespace arrow {
  nostd::shared_ptr<Span> GetCurrentSpan() {
    return ::arrow::internal::tracing::GetTracer()->GetCurrentSpan();
  }
}
```